cli/tests/common/test_environment.rs

```diff
@@ -120,6 +120,14 @@ impl TestEnvironment {
         // executables like `git` from the PATH.
         cmd.env("PATH", std::env::var_os("PATH").unwrap_or_default());
         cmd.env("HOME", self.home_dir.to_str().unwrap());
+        // Prevent git.subprocess from reading outside git config
+        let null_device = if cfg!(windows) {
+            "NUL"
+        } else {
+            "/dev/null"
+        };
```

Comment:
Does it work to use an empty string on all platforms?

Reply:
No, same problem.

I wonder if it has something to do with `\r\n` vs `\n`. `test_git_remote_with_branch_config` adds to the config file with `writeln!`, which [always just uses LF](https://doc.rust-lang.org/std/macro.writeln.html). If the beginning of the file uses `\r\n` but then later switches to `\n`, maybe that confuses Gitoxide? But then why would setting the env variables affect that 🤔.

Let me try and see if writing `\r\n` fixes it.

Reply:
Huh, that worked (even on non-Windows platforms). It feels like a bug/edge case in Gitoxide? I would report it but I don't have a Windows box to produce a minimal example.

Reply:
Our test might depend on the system git config (which would probably use LF on Windows), and gix might choose the platform-native line ending if there was no existing config files to derive style from. And gix might be strict about line ending chars when reformatting the config file.

Anyway, I think we can filter out uninteresting lines in tests.